verify.rb

```diff
@@ -24,60 +35,66 @@ def error(msg)
     puts "#{@output}. #{msg}"
   end
 
-  # Verify that the tfa factors are booleans
-  def check_tfa(website)
-    tfa = website['tfa']
-    if tfa != true && tfa != false
-      error("#{website['name']} \'tfa\' tag should be either \'Yes\' or \'No\'. (#{tfa})")
-    end
-
-    @tfa_forms.each do |tfa_form|
-      form = website[tfa_form]
-      next if form.nil?
-      unless website['tfa']
-        error("#{website['name']} should not contain a \'#{tfa_form}\' tag when it doesn\'t support TFA.")
+  # Validate an individual YAML tag
+  def check_tag(tag, required, tfa_state, website, onlyTrue = false)
+    if website[tag].nil?
+      if website['tfa'] == tfa_state && required
```

Comment:
maybe i've been out of the loop for a while, but this logic is a bit confusing - so this is not "tag is required" but "tag is required only if tfa flag is set"?